firmware/examples/Dac/source/main.cpp

```diff
@@ -0,0 +1,160 @@
+{
+    SJ2_ASSERT_FATAL(delay < 500,
+                     "Input frequency out of range! Frequency "
+                     "input needs to be between 0 and 500 Hz.");
+    for ( uint8_t i = 0; i < 255; i++ )
+    {
+        dac_demo->WriteDac(kSineLookup[i]);
+        Delay(delay);
+    }
+}
+void SawtoothDac(Dac * dac_demo, uint32_t delay)
+{
+    SJ2_ASSERT_FATAL(delay < 500,
+                     "Input delay out of range! Delay "
+                     "input needs to be between 0 and 500 Hz.");
+    int sawtooth = 0;
+    for ( uint8_t i = 0; i < 255; i++ )
+    {
+        uint16_t conversion = static_cast<uint16_t>(sawtooth);
+        dac_demo->WriteDac(conversion);
+        sawtooth = sawtooth + 4;
+        Delay(delay);
+    }
+}
+void TriangleDac(Dac * dac_demo, uint32_t delay)
+{
+    SJ2_ASSERT_FATAL(delay < 500,
+                     "Input delay out of range! Delay "
+                     "input needs to be between 0 and 500 Hz.");
+    int triangle = 0;
+    for ( uint8_t i = 0; i < 127; i++ )
+    {
+        uint16_t conversion = static_cast<uint16_t>(triangle);
+        dac_demo->WriteDac(conversion);
+        triangle = triangle + 8;
+        Delay(delay);
+    }
+    for ( uint8_t i = 127; i < 255 ; i++ )
+    {
+        uint16_t conversion = static_cast<uint16_t>(triangle);
+        dac_demo->WriteDac(conversion);
+        triangle = triangle - 8;
+        Delay(delay);
+    }
+}
+void SerratedDac(Dac * dac_demo, uint32_t delay)
+{
+    SJ2_ASSERT_FATAL(delay < 500,
+                     "Input delay out of range! Delay "
+                     "input needs to be between 0 and 500.");
+    int serrated = 0;
+    for ( uint8_t i = 0; i < 127; i++ )
+    {
+        uint16_t conversion = static_cast<uint16_t>(serrated);
+        dac_demo->WriteDac(conversion);
+        serrated = serrated + 8;
+        Delay(delay);
+    }
+    serrated = 0;
+    for ( uint8_t i = 127; i < 253 ; i++ )
+    {
+        uint16_t conversion = static_cast<uint16_t>(serrated);
+        dac_demo->WriteDac(conversion);
+        serrated = serrated + 4;
+        Delay(delay);
+    }
+}
+void StartDemo(Dac * dac_demo, uint32_t input_cycles)
+{
+    for (int i = 0; i < 50; i++)
+    {
+        for (int j = 0; j < 60; j++)
+        {
+            SineDac(dac_demo, input_cycles);
+        }
+        for (int j = 0; j < 60; j++)
+        {
+            SawtoothDac(dac_demo, input_cycles);
+        }
+        for (int j = 0; j < 60; j++)
+        {
+            TriangleDac(dac_demo, input_cycles);
+        }
+        for (int j = 0; j < 60; j++)
+        {
+            SerratedDac(dac_demo, input_cycles);
+        }
+    }
+}
+int main(void)
+{
+    #if SJ2_ENABLE_ANSI_CODES
+    // Clears the terminal
+    fputs("\e[2J\e[H", stdout);
+    #endif
+    fputs(SJ2_BACKGROUND_GREEN
+         "================================== SJTwo Booted! "
+         "==================================\n" SJ2_COLOR_RESET, stdout);
+    Dac * dac = new Dac();
```

Comment:
Even though this is not apart of the library, it is apart of the demos, which are apart of the library, which means do not allocate memory dynamically. Replace this with:

    Dac dac;